services/platform/platform.go

```diff
@@ -311,3 +313,58 @@ func (s *Service) disconnect(ctx context.Context, params *disconnectPMMParams) e
 
 	return nil
 }
+
+// SearchOrganizationTickets fetches the list of ticket associated with the Portal organization this PMM server is registered with.
+func (s *Service) SearchOrganizationTickets(ctx context.Context, req *platformpb.SearchOrganizationTicketsRequest) (*platformpb.SearchOrganizationTicketsResponse, error) {
+	userAccessToken, err := s.grafanaClient.GetCurrentUserAccessToken(ctx)
+	if err != nil {
+		if errors.Is(err, services.ErrFailedToGetToken) {
+			return nil, status.Error(codes.FailedPrecondition, "Failed to get access token. Please sign in using your Percona Account.")
+		}
+		s.l.Errorf("SearchOrganizationTickets request failed: %s", err)
+		return nil, internalServerError
+	}
+
+	ssoDetails, err := models.GetPerconaSSODetails(ctx, s.db.Querier)
+	if err != nil {
+		s.l.Errorf("failed to get SSO details: %s", err)
+		return nil, status.Error(codes.Aborted, "PMM server is not connected to Portal")
+	}
+
+	// Since PMM doesn't store the orgID we leave it empty and let Portal figure it out
+	// using the perconaPortal.orgID claim in the access token.
+	endpoint := fmt.Sprintf("https://%s/v1/orgs/%s/tickets:search", s.host, ssoDetails.PortalOrganizationID)
+
+	r, err := http.NewRequestWithContext(ctx, http.MethodPost, endpoint, nil)
+	if err != nil {
+		s.l.Errorf("Failed to build SearchOrganizationTickets request: %s", err)
+		return nil, internalServerError
+	}
+
+	h := r.Header
+	h.Add("Authorization", fmt.Sprintf("Bearer %s", userAccessToken))
+
+	resp, err := s.client.Do(r)
+	if err != nil {
+		s.l.Errorf("SearchOrganizationTickets request failed: %s", err)
+		return nil, internalServerError
+	}
+	defer resp.Body.Close() //nolint:errcheck
+
+	decoder := json.NewDecoder(resp.Body)
+	if resp.StatusCode != http.StatusOK {
```

Comment:
according to the spec all users (admin and technical) have access to this data, do we still need a check?  https://confluence.percona.com/display/PMM/Organization+Management+Service

Reply:
This spec defines roles on Portal level, but the tickets are fetched from ServiceNow (that has another roles levels). It may appear that user is not allowed to have access to tickets in ServiceNow and the response from Portal will be empty. 

Or another case - new role will appear on Portal that doesn't assume access to tickets at all. as a result there will be 'Permission Denied' error from Portal

Reply:
the second case is already handled in case we get permission denied from portal then the same error is returned by PMM too

for the first case there are two instances where portal returns empty response. one is when SN is not connected to Portal Org the other is when `ErrUserNotFound` user is not found on service now so I am not sure which error to return in this case.

According to the current code it will not return any error, just an empty response but you are right we should display some error

would 404 be appropriate here?

Reply:
@maxkondr for cases such as the SN not being linked to Portal org or SN maybe linked but the current user does not have a SN account Portal returns 200 with empty response, i don't think we should handle this any differently on PMM, because if we decide to show an error on PMM for any of these conditions there will be an inconsistency between saas-ui and PMM UI

for other errors from Portal such as Permission Denied, the error message are quite descriptive 
https://github.com/percona-platform/orgd/blob/main/service/orgd/service.go#L821
https://github.com/percona-platform/orgd/blob/main/service/orgd/service.go#L832

i am not sure what else to add here when showing these errors on PMM

> Or another case - new role will appear on Portal that doesn't assume access to tickets at all. as a result there will be 'Permission Denied' error from Portal

if this happens in the future in this case Portal will return some sort of error like `status.Error(codes.PermissionDenied, "User with Role X does not have the permission to view tickets")` and pmm-managed would forward the same error to PMM UI https://github.com/percona/pmm-managed/pull/966/files#diff-046eff8c6fd93bc9a6ace012834425228856532b375e8b7e218fea20f284a88dR377